src/java.base/share/classes/sun/security/pkcs/PKCS8Key.java

```diff
@@ -92,13 +92,16 @@ protected PKCS8Key() { }
      * This method is also used by {@link #parseKey} to create a raw key.
      */
     protected PKCS8Key(byte[] input) throws InvalidKeyException {
-        decode(new ByteArrayInputStream(input));
+        try {
+            decode(new DerValue(input));
+        } catch (IOException e) {
+            throw new InvalidKeyException("IOException: " +
+                    e.getMessage());
```

Comment:
How about including the cause in the IKE? Also, I suggest an error message such as "unable to decode key".

Same comments for `X509Key`.

Reply:
Sure.